crates/sui-light-client/src/main.rs

```diff
@@ -0,0 +1,680 @@
+
+            // print summary epoch and seq
+            println!(
+                "Epoch: {} Seq: {}: {}",
+                summary.epoch(),
+                summary.sequence_number,
+                summary.end_of_epoch_data.is_some()
+            );
+
+            if summary.epoch() == target_epoch && summary.end_of_epoch_data.is_some() {
+                found_summary = Some(summary);
+                break;
+            }
+
+            if summary.epoch() <= target_epoch {
+                start = mid + 1;
+            } else {
+                end = mid;
+            }
+        }
+
+        if let Some(summary) = found_summary {
+            // Note: Do not write summary to file, since we must only persist
+            //       checkpoints that have been verified by the previous committee
+
+            // Add to the list
+            checkpoints_list.checkpoints.push(summary.sequence_number);
+            write_checkpoint_list(config, &checkpoints_list)?;
+
+            // Update
+            last_epoch = summary.epoch();
+            last_checkpoint_seq = summary.sequence_number;
+        }
+    }
+
+    Ok(())
+}
+
+async fn check_and_sync_checkpoints(config: &Config) -> anyhow::Result<()> {
+    sync_checkpoint_list_to_latest(config).await?;
+
+    // Get the local checkpoint list
+    let checkpoints_list: CheckpointsList = read_checkpoint_list(config)?;
+
+    // Load the genesis committee
+    let mut genesis_path = config.checkpoint_summary_dir.clone();
+    genesis_path.push(&config.genesis_filename);
+    let genesis_committee = Genesis::load(&genesis_path)?.committee()?;
+
+    // Check the signatures of all checkpoints
+    // And download any missing ones
+
+    let mut prev_committee = genesis_committee;
+    for ckp_id in &checkpoints_list.checkpoints {
+        // check if there is a file with this name ckp_id.yaml in the checkpoint_summary_dir
+        let mut checkpoint_path = config.checkpoint_summary_dir.clone();
+        checkpoint_path.push(format!("{}.yaml", ckp_id));
+
+        // If file exists read the file otherwise download it from the server
+        let summary = if checkpoint_path.exists() {
+            read_checkpoint(config, *ckp_id)?
+        } else {
+            // Download the checkpoint from the server
+            let summary = download_checkpoint_summary(config, *ckp_id).await?;
+            summary.clone().verify(&prev_committee)?;
+            // Write the checkpoint summary to a file
+            write_checkpoint(config, &summary)?;
+            summary
+        };
+
+        // Print the id of the checkpoint and the epoch number
+        println!(
+            "Epoch: {} Checkpoint ID: {}",
+            summary.epoch(),
+            summary.digest()
+        );
+
+        // Extract the new committee information
+        if let Some(EndOfEpochData {
+            next_epoch_committee,
+            ..
+        }) = &summary.end_of_epoch_data
```

Comment:
in what situation this is none?

Reply:
This is `None` is this checkpoint is in fact not an end-of-epoch checkpoint, which may happen if the list provided is incorrect and contains non end-of-epoch checkpoints.